### PR TITLE
fix: resolve PR #42 review comments

### PR DESCRIPTION
**Pull request by marek-chmielowski-blurify**
## Summary

Resolves all 10 unresolved review comments left on #42 (merged without addressing them).

- Removed superfluous comments that only restated what `AGENTS.md`/naming already documents (`game.ts`, `admin-player-activity.ts`)
- Deduped `GAME_PERFORMANCE_SORT_FIELDS` between the contract port and admin-console (import from `@openora/core/contracts` instead of redeclaring)
- Added a `currency` filter to the game-performance report instead of summing bet/win amounts across currencies
- Rewrote DAU/WAU/MAU to a single `LEFT JOIN` + `FILTER` instead of three correlated subqueries per generated day
- Fixed a retention off-by-one: the registration-day session no longer counts as a return; the window now starts the day after registration
- Replaced hardcoded snake_case SQL identifiers with Drizzle column references in the retention query
- Added a `TODO` documenting that `winAmount` is never set yet (`endRound` has no way to report a win), so revenue currently always equals volume
- Rewrote the `DrizzleAdminGameReporting`, `DrizzleAdminPlayerActivity`, and `BackofficeService.getGamePerformance` test suites against real Postgres instead of mocked query builders/ports

## Test plan

- [x] `pnpm verify` (types, lint, boundaries, shape, deprecations, unit tests, drift) - green
- [x] New/rewritten real-Postgres tests cover: DAU/WAU/MAU correctness, retention window fix, currency filter, zero-activity edge cases

### CODE REVIEW

File: packages/core/src/admin-console/service/backoffice.service.ts [hunk 1]
```diff
@@ -168,6 +168,7 @@ export class BackofficeService {
       dateFrom: filter.dateFrom ? new Date(filter.dateFrom) : undefined,
       dateTo: filter.dateTo ? new Date(filter.dateTo) : undefined,
       gameType: filter.gameType,
+      currency: filter.currency,
```
Comment (agniev-a-hub) [inline packages/core/src/admin-console/service/backoffice.service.ts:171]: Somewhere you check for undefined, somewhere you don't. Make sure it's not affecting the filtering logic.

File: packages/core/src/casino/gaming/admin-reporting.ts [hunk 2]
```diff
@@ -14,20 +14,24 @@ export class DrizzleAdminGameReporting implements AdminGameReporting {
...
       filter.dateTo ? lte(gameRound.startedAt, filter.dateTo) : undefined,
+      filter.currency ? eq(gameRound.currency, filter.currency) : undefined,
     ].filter(Boolean);
     const where = filter.gameType ? eq(game.gameType, filter.gameType) : undefined;
 
     const volume = sql<string>`coalesce(sum(${gameRound.betAmount}), 0)`;
     // GGR - can be negative when a game pays out more than it takes in over the range.
+    // TODO: winAmount is never set today - endRound has no way to report a win, so it
+    // stays at its schema default('0') and revenue always equals volume until round
+    // settlement actually computes payouts.
```
Comment (agniev-a-hub) [inline packages/core/src/casino/gaming/admin-reporting.ts]: If it's another ticket's scope - just remove. TODO overall is bad :(

**Review by agniev-a-hub — DISMISSED** [review]
Filtering is WARN.

File: packages/core/src/iam/adapters/admin-player-activity.ts [hunk 3]
```diff
@@ -94,10 +102,10 @@ export class DrizzleAdminPlayerActivity implements AdminPlayerActivity {
       returned_users as (
         select distinct cu.user_id
         from cohort_users cu
-        inner join ${session} s
-          on s.user_id = cu.user_id
-          and s.updated_at >= cu.registered_at
-          and s.updated_at <= cu.registered_at + interval '1 day' * ${windowDays}
+        inner join ${session}
+          on ${session.userId} = cu.user_id
```
Comment (klaudia-blazyczek-blurify) [inline packages/core/src/iam/adapters/admin-player-activity.ts:105]: Because `cu.registered_at` is a timestamp, `cu.registered_at + interval '1 day'` strictly requires 24 hours to pass. If a user registers at 11:00 PM and plays again at 9:00 AM the next calendar day, they won't be counted as retained. It is usually better to base retention on calendar days using `cu.cohort_date + interval '1 day'`.

File: packages/core/src/casino/gaming/admin-reporting.ts [hunk 4]
```diff
@@ -14,20 +14,24 @@ export class DrizzleAdminGameReporting implements AdminGameReporting {
...
+    // status/date/currency filters scope which ROUNDS count towards a game's metrics,
+    // so they live in the join condition, not WHERE - a game with zero completed rounds
+    // in range still appears with all-zero metrics rather than being dropped from the
     // report. `gameType` narrows which GAMES appear at all, so it stays in WHERE.
     const joinConditions = [
       eq(gameRound.gameId, game.id),
       eq(gameRound.status, 'completed'),
       filter.dateFrom ? gte(gameRound.startedAt, filter.dateFrom) : undefined,
       filter.dateTo ? lte(gameRound.startedAt, filter.dateTo) : undefined,
+      filter.currency ? eq(gameRound.currency, filter.currency) : undefined,
```
Comment (klaudia-blazyczek-blurify) [inline packages/core/src/casino/gaming/admin-reporting.ts:28]: If `currency` is not provided in the filter, this query will sum `betAmount` and `winAmount` across all currencies into a single mixed-currency aggregate. Should the currency filter be required, or should the query group by currency if it's missing?

File: packages/core/src/casino/gaming/admin-reporting.ts (hunk 2 above)
Comment (zaxovaiko) [reply packages/core/src/casino/gaming/admin-reporting.ts]: Why is TODO bad? 😄

File: packages/core/src/admin-console/service/backoffice.service.ts (hunk 1 above)
Comment (marek-chmielowski-blurify) [reply packages/core/src/admin-console/service/backoffice.service.ts:171]: filter.currency is already `string | undefined` (same shape as dateFrom/dateTo), and it's checked downstream in the join condition (`filter.currency ? eq(gameRound.currency, filter.currency) : undefined`), so passing it through unconditionally here is safe - an undefined currency correctly results in no currency filter being applied. Leaving as-is.

File: packages/core/src/casino/gaming/admin-reporting.ts (hunk 4 above)
Comment (marek-chmielowski-blurify) [reply packages/core/src/casino/gaming/admin-reporting.ts:28]: Intentional - when no currency filter is applied the report intentionally sums across currencies without FX conversion, and the operator-facing UI shows a disclaimer in that case so the number isn't misread as a converted total. Added a one-line comment in the code documenting this as a design decision.

File: packages/core/src/iam/adapters/admin-player-activity.ts (hunk 3 above)
Comment (marek-chmielowski-blurify) [reply packages/core/src/iam/adapters/admin-player-activity.ts:105]: Good catch - fixed. The join condition now uses cu.cohort_date + interval '1 day' instead of cu.registered_at + interval '1 day', so retention is based on calendar days rather than a strict 24h window. Also removed the now-unused registered_at column from the CTE and added a regression test covering exactly this case (registers 23:00, returns 09:00 the next calendar day).

**Review by klaudia-blazyczek-blurify — APPROVED** [review]
